src/core/collector.ml

```diff
@@ -53,6 +53,10 @@ module type S = sig
   (** Exit an explicit span.
       @since 0.3 *)
 
+  val add_data_to_span : span -> (string * user_data) list -> unit
+  (** @since Adds data to the given span.
+      NEXT_RELEASE *)
+
```

Comment:
See [my review](https://github.com/c-cube/ocaml-trace/pull/11#pullrequestreview-1601545377) of #11